frontend/src/common/ReviewWithComment.tsx

```diff
@@ -1,45 +1,123 @@
 
-export default function ReviewWithComment({ title, description, open, onClose, onSubmit }: ReviewWithCommentProps) {
+export default function ReviewWithComment({
+  title,
+  description,
+  open,
+  onClose,
+  onSubmit,
+  release,
+}: ReviewWithCommentProps) {
+  const { reviews } = useGetReviewRequestsForModel(release.modelId, release.semver, true)
+  const { modelRoles, isModelRolesLoading, isModelRolesError } = useGetModelRoles(reviews[0].model.id)
+
   const [reviewComment, setReviewComment] = useState('')
   const [showError, setShowError] = useState(false)
+  const [selectOpen, setSelectOpen] = useState(false)
+  const [reviewKind, setReviewKind] = useState<ReviewRequestInterface>()
+
+  useEffect(() => {
+    if (reviews) {
+      setReviewKind(reviews[0])
+    }
+  }, [reviews])
 
   function invalidComment() {
     return reviewComment.trim() === '' ? true : false
   }
 
   function submitForm(kind: ResponseTypeKeys) {
     setShowError(false)
-    if (invalidComment() && (kind === ResponseTypes.Reject || kind === ResponseTypes.RequestChanges)) {
+    if (invalidComment() && kind === ResponseTypes.RequestChanges) {
       setShowError(true)
     } else {
-      onSubmit(kind, reviewComment)
+      if (reviewKind) {
+        onSubmit(kind, reviewComment, reviewKind.role)
+      }
     }
   }
 
+  function onChange(_event: React.SyntheticEvent<Element, Event>, newValue: ReviewRequestInterface | null) {
+    if (newValue) {
+      setReviewKind(newValue)
+    }
+  }
+
+  if (isModelRolesError) {
+    return <MessageAlert message={isModelRolesError.info.message} severity='error' />
+  }
+
   return (
     <>
-      <Dialog open={open} onClose={onClose}>
+      {isModelRolesLoading && <Loading />}
+      <Dialog fullWidth open={open} onClose={onClose}>
         <DialogTitle>{title}</DialogTitle>
         <DialogContent>
           <Stack spacing={2}>
+            <Autocomplete
+              sx={{ pt: 1 }}
+              open={selectOpen}
+              onOpen={() => {
+                setSelectOpen(true)
+              }}
+              onClose={() => {
+                setSelectOpen(false)
+              }}
+              // we might get a string or an object back
+              isOptionEqualToValue={(option: ReviewRequestInterface, value: ReviewRequestInterface) =>
+                option.role === value.role
+              }
+              onChange={onChange}
+              value={reviewKind}
+              disabled={reviews.length === 1}
+              getOptionLabel={(option) => getRoleDisplay(option.role, modelRoles)}
+              options={reviews}
+              renderInput={(params) => (
+                <TextField
+                  {...params}
+                  label='Select your role'
+                  size='small'
+                  InputProps={{
+                    ...params.InputProps,
+                    endAdornment: <>{params.InputProps.endAdornment}</>,
```

Comment:
What's the point of this? It looks like a long-winded way of writing `InputProps={params.InputProps}` to me and, as we're already spreading `params` above, this whole `InputProps` part can probably be removed.